test/internal/k8sutils/utils.go

```diff
@@ -267,6 +267,47 @@ func WaitForPodDeployment(ctx context.Context, clientset *kubernetes.Clientset,
 	return errors.Wrapf(retrier.Do(ctx, checkPodDeploymentFn), "could not wait for deployment %s", deploymentName)
 }
 
+func WaitForPodDaemonset(ctx context.Context, clientset *kubernetes.Clientset, namespace, daemonsetName, podLabelSelector string) error {
+	podsClient := clientset.CoreV1().Pods(namespace)
+	daemonsetClient := clientset.AppsV1().DaemonSets(namespace)
+	checkPodDaemonsetFn := func() error {
+		daemonset, err := daemonsetClient.Get(ctx, daemonsetName, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "could not get daemonset %s", daemonsetName)
+		}
+
+		if daemonset.Status.NumberReady == 0 && daemonset.Status.DesiredNumberScheduled == 0 {
+			// Capture daemonset restart. Restart sets every numerical status to 0.
+			log.Printf("daemonset %s has %d pods in ready status, expected %d", daemonsetName, daemonset.Status.NumberReady, daemonset.Status.DesiredNumberScheduled)
+			return errors.New("daemonset has not correctly scheduled the right number of pods")
```

Comment:
I am thinking whether we should use all errors.Wraf() here?

Reply:
The errors are originating from here. We shouldn't use wrapf() as it will capture the empty value from any error call before it.